### PR TITLE
Doc:Add parquet,orc,avro delete codec and level

### PR DESCRIPTION
**Pull request by renshangtao**
Add the parquet,orc,avro delete codec and level properties in configuration

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -72,7 +82,7 @@ Iceberg tables support table properties to configure table behavior, like the de
 | write.wap.enabled                  | false              | Enables write-audit-publish writes |
 | write.summary.partition-limit      | 0                  | Includes partition-level summary stats in snapshot summaries if the changed partition count is less than this limit |
 | write.metadata.delete-after-commit.enabled | false      | Controls whether to delete the oldest version metadata files after commit |
-| write.metadata.previous-versions-max       | 100        | The max number of previous version metadata files to keep before deleting after commit |
+| write.metadata.previous-versions-max      | 100        | The max number of previous version metadata files to keep before deleting after commit |
```
Comment (rdblue) [inline docs/configuration.md:85]: Can you revert this whitespace-only change?

File: docs/configuration.md [hunk 2]
```diff
@@ -52,15 +52,25 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | write.parquet.dict-size-bytes      | 2097152 (2 MB)     | Parquet dictionary page size                       |
+| write.delete.parquet.dict-size-bytes | 2097152 (2 MB)   | Parquet dictionary page size                       |
 | write.parquet.compression-codec    | gzip               | Parquet compression codec: zstd, brotli, lz4, gzip, snappy, uncompressed |
 | write.parquet.compression-level    | null               | Parquet compression level                          |
+| write.delete.parquet.compression-codec | gzip           | Parquet compression codec: zstd, brotli, lz4, gzip, snappy, uncompressed |
+| write.delete.parquet.compression-level | null           | Parquet compression level                          |
 | write.avro.compression-codec       | gzip               | Avro compression codec: gzip(deflate with 9 level), zstd, snappy, uncompressed |
 | write.avro.compression-level       | null               | Avro compression level                             |
+| write.delete.avro.compression-codec | gzip              | Avro compression codec: gzip(deflate with 9 level), zstd, snappy, uncompressed |
```
Comment (rdblue) [inline docs/configuration.md:64]: Clauses within parens should be separated from the preceding words by a space: `gzip (deflate ...)`.

File: docs/configuration.md [hunk 3]
```diff
@@ -52,15 +52,25 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| write.delete.avro.compression-level | null              | Avro compression level                             |
 | write.orc.stripe-size-bytes        | 67108864 (64 MB)   | Define the default ORC stripe size, in bytes       |
+| write.delete.orc.stripe-size-bytes | 67108864 (64 MB)   | Define the default ORC stripe size, in bytes       |
 | write.orc.block-size-bytes         | 268435456 (256 MB) | Define the default file system block size for ORC files |
+| write.delete.orc.block-size-bytes  | 268435456 (256 MB) | Define the default file system block size for ORC files |
 | write.orc.compression-codec        | zlib               | ORC compression codec: zstd, lz4, lzo, zlib, snappy, none |
 | write.orc.compression-strategy     | speed              | ORC compression strategy: speed, compression |
+| write.delete.orc.compression-codec | zlib               | ORC compression codec: zstd, lz4, lzo, zlib, snappy, none |
+| write.delete.orc.compression-strategy | speed           | ORC compression strategy: speed, compression |
```
[the remark this answers is not in the document]
Comment (renshangtao) [reply docs/configuration.md]: yes the delete is the same as write.orc.com press-codec

File: docs/configuration.md (hunk 1 above)
Comment (renshangtao) [reply docs/configuration.md]: Thank you for your reply. I have had revert the whitespace

File: docs/configuration.md (hunk 2 above)
Comment (renshangtao) [reply docs/configuration.md]: I have modified it.

**Comment by hililiwei** [comment]
Although we can configure these for the DELETE separately, I do not recommend exposing them.
If we must do so, its default value is consistent with data file. 
e.g.:
        `long stripeSize = PropertyUtil.propertyAsLong(config, DELETE_ORC_STRIPE_SIZE_BYTES, dataContext.stripeSize());`
So, maybe it should look like this:
| Property | Default | Description |
| ---------------------------------- | ------------------ | -------------------------------------------------- |
| write.delete.orc.compression-codec | write.orc.compression-codec | ORC compression codec: zstd, lz4, lzo, zlib, snappy, none |

**Comment by renshangtao** [comment]
@hililiwei Yes, you are right.The parquet and avro is the same as orc.
``
String codecAsString = config.get(DELETE_PARQUET_COMPRESSION);
``
``
CompressionCodecName codec = codecAsString != null ? toCodec(codecAsString) : dataContext.codec();
``

@rdblue What is your opinion? Do we need to expose them.

Can we modify it like this
Property | Default | Description
-- | -- | --
write.delete.orc.compression-codec | data compression codec | ORC compression codec: zstd, lz4, lzo, zlib, snappy, none



In the current document the write.delete format.default was exposed

Property | Default | Description
-- | -- | --
write.format.default | parquet | Default file format for the table; parquet, avro, or orc
write.delete.format.default | data file format | Default delete file format for the table; parquet, avro, or orc

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
I think it is good to note that the default is the data value, but document the settings exist.

**Pull request closed without merging**